### PR TITLE
Many Changes

### PR DESCRIPTION
**Pull request by lesv**
1. remove eclipse profiles (we should either support everything or none)
2. add index.yaml to 2-structured-data
3. Use new plugins
4. Fix SQL
5. Fix list by user
6. Propagate all fixes to next version.
7. change paths (made shorter, removed managedvms)

Still to do - remove personalized data; remove flex keywords;  Add some
tags for docs;  Lots of docs fixes.

### CODE REVIEW

**Comment by lesv** [comment]
8. rename setters / getters for updated google-cloud-java

**Comment by lesv** [comment]
@tswast PTAL - I hate reviewing things this long -- apologies for giving it to you.  I'll probably need to update travis.sh for this as well.

**Comment by tswast** [comment]
That get/set/new change to the Google Cloud libraries sure made for a lot of changes. Are we sure it was worth it? (I know it's done so it's not like we'd switch back)

File: bookshelf/2-structured-data/src/main/java/com/example/getstarted/basicactions/ListBookServlet.java [hunk 1]
```diff
@@ -50,11 +49,11 @@ public void init() throws ServletException {
         break;
       case "cloudsql":
         try {
-          if (System.getenv().containsKey("GAE_MODULE_INSTANCE")) {
-            dao = new CloudSqlDao(this.getServletContext().getInitParameter("sql.urlRemote"));
-          } else {
-            dao = new CloudSqlDao(this.getServletContext().getInitParameter("sql.urlLocal"));
+          String connect = this.getServletContext().getInitParameter("sql.urlRemote");
```
Comment (lesv) [inline bookshelf/2-structured-data/src/main/java/com/example/getstarted/basicactions/ListBookServlet.java]: Yes

File: bookshelf/6-gce/src/main/java/com/example/getstarted/objects/Book.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+    this.createdBy = builder.createdBy;
+    this.createdById = builder.createdById;
+    this.publishedDate = builder.publishedDate;
+    this.description = builder.description;
+    this.id = builder.id;
+    this.imageUrl = builder.imageUrl;
+  }
+// [END constructor]
+// [START builder]
+  public static class Builder {
```
Comment (lesv) [inline bookshelf/6-gce/src/main/java/com/example/getstarted/objects/Book.java]: Eamonn didn't think laying AutoValue on users at this point was a good idea.  I was good w/ the idea.

File: bookshelf/5-logging/src/main/java/com/example/getstarted/daos/DatastoreDao.java [hunk 3]
```diff
@@ -118,15 +117,15 @@ public void deleteBook(Long bookId) {
     if (startCursorString != null && !startCursorString.equals("")) {
       startCursor = Cursor.fromUrlSafe(startCursorString);    // Where we left off
     }
-    Query<Entity> query = Query.entityQueryBuilder()          // Build the Query
-        .kind("Book6")                                         // We only care about Books
-        .limit(10)                                            // Only show 10 at a time
-        .startCursor(startCursor)                             // Where we left off
-        .orderBy(OrderBy.asc(Book.TITLE))                     // Use default Index "title"
+    Query<Entity> query = Query.newEntityQueryBuilder()       // Build the Query
+        .setKind("Book5")                                     // We only care about Books
```
Comment (lesv) [inline bookshelf/5-logging/src/main/java/com/example/getstarted/daos/DatastoreDao.java]: Yep or some variable that comes from a POM or somewhere else -- this whole idea has issues.

**Comment by lesv** [comment]
ADMIN override for this.  I've built it a lot.

**Comment by lesv** [comment]
Actually, I want to prep a CL to match this PR.

**Pull request merged**